prospero-common/src/main/java/org/wildfly/prospero/actions/ApplyCandidateAction.java

```diff
@@ -125,6 +131,35 @@ public ApplyCandidateAction(Path installationDir, Path updateDir)
         }
     }
 
+    private ChannelSession getChannelSession(Path installationDir) throws ProvisioningException, OperationException {
+        final MavenOptions mavenOptions = MavenOptions.builder()
+                .setOffline(true)
+                .setNoLocalCache(true)
+                .build();
+        try (GalleonEnvironment galleonEnv = GalleonEnvironment.builder(installationDir,
+                        InstallationMetadata.loadInstallation(installationDir).getProsperoConfig().getChannels(),
+                        new MavenSessionManager(mavenOptions))
+                .build()) {
+
+            return galleonEnv.getChannelSession();
+        }catch (UnresolvedChannelMetadataException | MetadataException e){
+            log.debug(e.getMessage());
+            return GalleonEnvironment.builder(installationDir,
+                    Collections.emptyList(),
+                    new MavenSessionManager(mavenOptions)).build().getChannelSession();
+        }
+    }
+    private String getChannelName(Artifact artifact) {
```

Comment:
I'm not sure about if this would work. 
1. The channel name might not be possible to be resolved in offline 
2. Between creating the candidate and applying it new versions of artifacts could be published possibly into different channels then listed here. 

The better solution would be to use UpdateSet generated during the building of the candidate to generate a record of updates in the candidate and use that information in apply candidate step.

Reply:
> I'm not sure about if this would work.
> 
> 1. The channel name might not be possible to be resolved in offline
> 2. Between creating the candidate and applying it new versions of artifacts could be published possibly into different channels then listed here.
> 
> The better solution would be to use UpdateSet generated during the building of the candidate to generate a record of updates in the candidate and use that information in apply candidate step.

Hi @spyrkob ,

I've updated the above changes, please let me know if any review changes are required.
Thanks.